### ISSUE TITLE
Window size includes shadow [$15] 

### ISSUE THREAD

**Issue by wezm** [elementary/stylesheet#80]
In a GTK+ 3.22 environment the window dimensions appear to include the shadow of the window. Thus a window can be brought to the foreground by clicking its shadow and the resize controls remain far beyond the edge of the window.

Note how the resize cursor is active for Slack far past the edge of the window:

![screenshot from 2017-03-23 21-47-45](https://cloud.githubusercontent.com/assets/21787/24244325/1cf7ff52-100a-11e7-9e4b-faefcc041398.png)

It seems to be worse for Qt and fake native apps (electron) like Slack.

---
There is a bounty on this issue [on Bountysource](https://www.bountysource.com/issues/43354129-window-size-includes-shadow-15)

**Comment by Defman21** [comment]
Does it happen with native apps though? (Like patheon-files)

**Comment by wezm** [comment]
Yes it happens with the native apps too, but to a lesser degree. I'm having trouble getting a screen shot of it but the cursor remains the resize cursor 5–7mm away from the edge of the window in pantheon-file and pantheon-terminal.

**Comment by cassidyjames** [comment]
Can confirm this happens with the elementary theme on Arch on Wayland with GTK 3.22. It might actually happen with other themes, but only be obvious because we have nice, big shadows. Probably requires more investigation to see what's up. I'm also not sure if it happens in X and Pantheon with 3.22.

**Comment by danrabbit** [comment]
Looks like in newer adwaita the cursor resize area is set by specifying `margin` on `.window-frame`

**Comment by wezm** [comment]
>  I'm also not sure if it happens in X and Pantheon with 3.22.

I'm using pantheon on Arch with X.org and it happens there too.

**Comment by Conan-Kudo** [comment]
Confirmed in Fedora 25 and 26, too.

**Comment by danrabbit** [comment]
Please don't make +1 comments, just use the thumbs up reaction

**Comment by danrabbit** [comment]
I can confirm this issue with slack but not with other apps that use `default-decoration` (like startup disk creator). My guess is that this is actually a bug in Gala as changing `margin` seems to have no effect on these windows

**Comment by felipunkz** [comment]
Guys I know this issue is opened for elementary but I can confirm the exact same behavior on Gnome (at least Ubuntu Gnome 17.04) if you have any news about it I'd like to subscribe to whatever solution you find, Thanks!
![selection_073](https://user-images.githubusercontent.com/5734716/30780109-dc738618-a0c8-11e7-975c-10ff9fb13614.png)

**Comment by Suzamax** [comment]
Please notice that the problem has nothing to do with the CSS but it's a **Gala** issue. `.ssd` property only affects the borders, and it's **Gala** who creates the actual shadows for the server-side decorations.

Problem is _probably_ here: https://github.com/elementary/gala/blob/master/src/ShadowEffect.vala

CSD applications use Mutter instead of Vala. PROOF:

![captura realizada el 2017-11-16 17 52 31](https://user-images.githubusercontent.com/10341605/32904163-23ad2ac2-caf7-11e7-94ee-628b920ab92c.png)

I hope this may help you found out the bug.

Source: Adwaita:
```
// server-side decorations as used by mutter
.ssd & { box-shadow: 0 0 0 1px $_wm_border; } //just doing borders, wm draws actual shadows
```

**Comment by danrabbit** [comment]
This was fixed in #327 :)
